plugins/outputs/sql/sql.go

```diff
@@ -98,11 +98,18 @@ func (p *SQL) Init() error {
 }
 
 func (p *SQL) Connect() error {
-	db, err := gosql.Open(p.Driver, p.DataSourceName)
+	secretBuffer, err := p.DataSourceName.Get()
+	if err != nil {
+		return fmt.Errorf("loading data source name secret failed: %w", err)
+	}
+
+	db, err := gosql.Open(p.Driver, secretBuffer.String())
 	if err != nil {
 		return fmt.Errorf("creating database client failed: %w", err)
 	}
 
+	secretBuffer.Destroy()
```

Comment:
This is dangerous as it might leave a locked buffer behind in case we run into the error case! Either use a `defer` statement to destroy the buffer _or_ (better) store the string in a local variable and directly destroy the buffer. Same for the cases below. ..